contracts/recovery-contracts/RecoveryForwarder.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity ^0.4.18;
+
+/**
+ * Contract that will forward any incoming Ether to the creator of the contract
+ */
+contract RecoveryForwarder {
+  // Address to which any funds sent to this contract will be forwarded
+  address public signer;
+  /**
+   * Create the contract, and sets the destination address to that of the creator
+   */
+  function init(address _signer) external {
+    signer = _signer;
+  }
+
+   /**
+   * Modifier that will execute internal code block only if the sender is an authorized signer on this wallet
+   */
+  modifier onlySigner {
+    if (signer != msg.sender) {
+      revert();
+    }
+    _;
+  }
+
+   /**
+   * Default function; Gets called when Ether is deposited
+   */
+  function() public payable {
+  }
+
+  /**
+   * Execute a transaction from this wallet using the signer.
+   *
+   * @param toAddress the destination address to send an outgoing transaction
+   * @param value the amount in Wei to be sent
+   * @param data the data to send to the toAddress when invoking the transaction
+   */
+  function send(
+      address toAddress,
+      uint value,
+      bytes data
+  ) external onlySigner {
+    // Success, send the transaction
+    if (!(toAddress.call.value(value)(data))) {
+      // Failed executing transaction
+      revert();
+    }
+  }
```

Comment:
We use this function to transfer native coin and tokens which is controlled by signer